torch/csrc/autograd/functions/convolution.cpp

```diff
@@ -328,11 +357,19 @@ auto ConvBackward::apply(const variable_list& grad_outputs) -> variable_list {
     }
   }
 
+  // Add saved variables used out of the pure autograd to inputs
+  variable_list all_inputs(grad_outputs);
+  all_inputs.push_back(input_.unpack());
+  all_inputs.push_back(weight_.unpack());
+
   auto outputs =  as_tensor_list(std::move(grad_input),
                                  std::move(grad_weight),
                                  std::move(grad_bias));
-  return wrap_outputs(grad_outputs, std::move(outputs), [&](FunctionFlags f) {
-    return std::make_shared<Error>("ConvBackward is not differentiable", std::move(f));
+  return wrap_outputs(all_inputs, std::move(outputs), [&](FunctionFlags f) {
+    return std::make_shared<ConvBackwardBackward>(
+      f, *this,
+      input_.unpack()->save(this), weight_.unpack()->save(this),
```

Comment:
Unpacking does a bunch of checks and it's best to do it only once. You can change the beginning of the function to use `unpack()` to get a Variable, and extract its data from the member. Then, reuse the unpacked Variables in here